soc/xtensa/intel_adsp/tools/cavstool_client.py

```diff
@@ -50,18 +54,51 @@ def send_cmd(self, cmd):
 
     def download(self, filename):
         # Send the FW to server
+        fname = os.path.basename(filename)
+        fsize = os.path.getsize(filename)
+        flen = len(fname)
+
+        md5_tx = hashlib.md5(open(filename,'rb').read()).hexdigest()
+
+        # Pack the header and the expecting packed size is 80 bytes.
+        # The header by convention includes:
+        # size(2), filename(42), MD5(32) and pad(2)
+        values = (fsize, fname.encode('utf-8'), md5_tx.encode('utf-8'), flen)
+        log.info(f'filename:{fname}, size:{fsize}, md5:{md5_tx}, len {flen}')
+
+        s = struct.Struct('I 42s 32s H')
+        packed_data = s.pack(*values)
+        packed_size = s.size
+        log.info(f'packed size: {packed_size}')
+
         with open(filename,'rb') as f:
-            log.info('Sending...')
-            ret = self.sock.sendfile(f)
+            log.info(f'Sending...')
+
+            m = packed_data + f.read(BUF_SIZE - packed_size)
+            self.sock.send(m)
+            ret = len(m)
+
+            while True:
+                l = f.read(1024)
+                if not l:
+                    break
+
+                self.sock.send(l)
+                ret += len(l)
+
             log.info(f"Done Sending ({ret}).")
 
+            rck = self.sock.recv(BUF_SIZE).decode("utf-8")
+            log.info(f"RCK ({rck}).")
+            if not rck == "success":
+                global RET
+                RET = -1
+
     def run(self):
         filename = str(self.args.fw_file)
-        send_fn = os.path.basename(filename)
-        self.sock.sendall(send_fn.encode("utf-8"))
-        log.info(f"Sent fw:     {send_fn}, {filename}")
 
         self.download(filename)
+        log.info(f"Done Transfer")
```

Comment:
Why this extra message? And why no message a few lines above, in the case the `success` reply is not received?

Reply:
Ok, I remove it here and put a log.error in above code where did not receive successfully. You are right, thanks.